packages/dd-trace/test/plugins/plugin-structure.spec.js

```diff
@@ -0,0 +1,104 @@
+'use strict'
+
+require('../setup/tap')
+
+const { expect } = require('chai')
+
+const fs = require('fs')
+const path = require('path')
+
+const abstractPlugins = [
+  'web' // web is an abstract plugin, and will not have an instrumentation file
+]
+
+describe('Plugin Structure Validation', () => {
+  const packagesDir = path.join(__dirname, '..', '..', '..')
+  const instrumentationsDir = path.join(packagesDir, 'datadog-instrumentations', 'src')
+
+  let pluginDirs
+  let allPluginIds
+  let instrumentationFiles
+
+  before(() => {
+    pluginDirs = fs.readdirSync(packagesDir)
+      .filter(dir => dir.startsWith('datadog-plugin-'))
+      .filter(dir => {
+        const pluginPath = path.join(packagesDir, dir, 'src', 'index.js')
+        return fs.existsSync(pluginPath)
```

Comment:
How do we handle the case when a plugin has no such file? There is no id as such, if I am not mistaken but how do we get that instead?

Reply:
I have made the test more explicit, and added an allowlist of a few plugin directories which exist (`axios`, `limitd-client`, `mongoose`, which we have plugin dirs for but no actual plugin implemented). I also added comments explaining why they exist. These are special cases and should not be the norm.